clib/cList.ml

```diff
@@ -315,6 +315,15 @@ let map2 f l1 l2 = match l1, l2 with
     cast c
   | _ -> invalid_arg "List.map2"
 
+let rec concat_map_loop f p = function
+  | [] -> ()
+  | x :: l -> concat_map_loop f (copy p (f x)) l
+
+let concat_map f l =
+  let dummy = { head = Obj.magic 0; tail = [] } in
+  concat_map_loop f dummy l;
+  dummy.tail
```

Comment:
If this is provided by some OCaml > 4.09, maybe adding a comment like `(* remove when requiring OCaml >= 4.xx *)` would help cleaning the day we'll raise the upper bound on OCaml.

Reply:
It depends if you want to remove operations on lists when they merely become provided by the Stdlib, or when they are as fast as this implementation (basically when the tail mod cons versions are used). In the former case, the bound is `4.10.0` while in the latter case it is `5.1.0`. I don't think my uses need a fast version, but I don't know for the future uses

Reply:
Might be safer to mention 5.1.0 then.

Reply:
I can add it, but all functions which are reimplementations of stdlib functions using `'a cell` will be as redundant once 5.1.0 and its tmc list operations are required